shell/common/platform_view.h

```diff
@@ -51,13 +51,17 @@ namespace flutter {
 ///
 class PlatformView {
  public:
+  using AddViewCallback = std::function<void(bool added)>;
```

Comment:
What do the `added` and `removed` arguments to the callback mean? If `added` is false, why invoke the `AddViewCallback`? Perhaps document that?

Reply:
Both the add and the remove callbacks rely on the result. For example, in the Windows embedder as implemented in https://github.com/flutter/engine/pull/51923
* The embedder blocks on the `AddView` callback until a result is reported, which decides whether the returned `FlutterWindowsView` is null.
* The embedder also pre-allocate resources after calling `AddView` before the callback, and if the result turns out a failure, then the resources are deallocated. The pre-allocation ensures that the framework can start operating the view as soon as they're aware of it.